nemo_skills/inference/generate.py

```diff
@@ -502,7 +519,14 @@ async def process_single_datapoint(self, data_point, all_data):
             if self.cfg.override_max_code_executions and self.cfg.total_code_executions_in_prompt is not None:
                 generation_params["max_code_executions"] = data_point["total_code_executions"]
 
-        return await self.llm.generate_async(**generation_params)
+        result = await self.llm.generate_async(**generation_params)
+
+        # Apply evaluation hook if configured
+        if self.evaluator:
+            eval_results = await self.evaluator.eval_single({**data_point, **result})
```

Comment:
this might be a bit tricky as e.g. it ignores the generation_key so it might be hard to reuse logic between batch and single evaluation. We can leave this as is I guess but might need to refactor things a bit so that we can ensure the structure of the final data points is identical between batch and single evals

Reply:
can you add a TODO above about this?